test/hotspot/jtreg/TEST.groups

```diff
@@ -26,11 +26,19 @@
 all = \
   :hotspot_all
 
+# The resourcehos tests might require a lot of resources  and
+# shouldn't be executed concurrently with ANY other tests
+# So the group shouldn't be part of any other group
+hotspot_resourcehogs = \
+  resourcehogs
+
 hotspot_all = \
+  -:hotspot_resourcehogs \
```

Comment:
When "all" was defined it was specifically intended to be all. If you want all-but-something then you define a group that is all-but -something. I don't know if these groups even get used regularly by anyone.

Reply:
There is a problem with resource_hogs, it is a specific group that might affect any other tests. It is not safe to execute :hotspot_resourcehogs with any other tests. I think that we don't have this problem know or we might have just rare intermittent failures/timeouts of random tests. 
Do we need any groups that are potentially unsafe at all? I don't think so and consider these tests like '/manually' that should be automatically ignored .

Reply:
It is up to the people running "all" (see in particular JDK-8323515) to decide what makes sense, IMO. If we are running "all" and have an issue then we should change what we run.